File: Classes/EVEOnlineAPI/EVE/EVEFacWarStats.m
//
//  EVEFacWarStats.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 8/17/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "EVEFacWarStats.h"

@implementation EVEFacWarStatsKills

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"killsYesterday":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"killsLastWeek":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"killsTotal":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"victoryPointsYesterday":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"victoryPointsLastWeek":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"victoryPointsTotal":@{@"type":@(EVEXMLSchemePropertyTypeScalar)}};
	return scheme;
}

@end


@implementation EVEFacWarStatsFactionsItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme) {
		scheme = @{@"factionID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"factionName":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"pilots":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"systemsControlled":@{@"type":@(EVEXMLSchemePropertyTypeScalar)}};
		
		NSMutableDictionary* dic = [[super scheme] mutableCopy];
		[dic setValuesForKeysWithDictionary:scheme];
		scheme = dic;
	}
	
	return scheme;
}

@end


@implementation EVEFacWarStatsFactionWarsItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"factionID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"factionName":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"againstID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"againstName":@{@"type":@(EVEXMLSchemePropertyTypeString)}};
	return scheme;
}

@end


@implementation EVEFacWarStats

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"totals":@{@"type":@(EVEXMLSchemePropertyTypeObject), @"class":[EVEFacWarStatsKills class]},
				   @"factions":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarStatsFactionsItem class]},
				   @"factionWars":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEFacWarStatsFactionWarsItem class]}};
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVEFacWarStats.swift
//
//  EVEFacWarStats.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 29.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

public class EVEFacWarStats: EVEResult {
	public var factionID: Int64 = 0
	public var factionName: String = ""
	public var enlisted: Date = Date.distantPast
	public var currentRank: Int = 0
	public var highestRank: Int = 0
	public var killsYesterday: Int = 0
	public var killsLastWeek: Int = 0
	public var killsTotal: Int = 0
	public var victoryPointsYesterday: Int = 0
	public var victoryPointsLastWeek: Int = 0
	public var victoryPointsTotal: Int = 0
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
		"factionID":EVESchemeElementType.Int64(elementName: nil, transformer: nil),
		"factionName":EVESchemeElementType.String(elementName: nil, transformer: nil),
		"enlisted":EVESchemeElementType.Date(elementName: nil, transformer: nil),
		"currentRank":EVESchemeElementType.Int(elementName: nil, transformer: nil),
		"highestRank":EVESchemeElementType.Int(elementName: nil, transformer: nil),
		"killsYesterday":EVESchemeElementType.Int(elementName: nil, transformer: nil),
		"killsLastWeek":EVESchemeElementType.Int(elementName: nil, transformer: nil),
		"killsTotal":EVESchemeElementType.Int(elementName: nil, transformer: nil),
		"victoryPointsYesterday":EVESchemeElementType.Int(elementName: nil, transformer: nil),
		"victoryPointsLastWeek":EVESchemeElementType.Int(elementName: nil, transformer: nil),
		"victoryPointsTotal":EVESchemeElementType.Int(elementName: nil, transformer: nil),
		]
	}
}

